packages/labs/ssr/src/lib/render-result-readable.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {Readable} from 'stream';
+import {isIterable, RenderResult} from './render-result.js';
+
+type RenderResultIterator = Iterator<
+  string | RenderResult | Promise<string | RenderResult>
+>;
+
+/**
+ * A Readable that reads from a RenderResult.
+ */
+export class RenderResultReadable extends Readable {
+  private _result: RenderResult;
+
+  /**
+   * A stack of open iterators.
+   *
+   * We need to keep this as instance state because we can pause and resume
+   * reading values at any time and can't run guarentee to run iterators to
```

Comment:
Something typo'd in this sentence, but not exactly sure what it should be ("cant run guarantee to run")